controllers/update_status_test.go

```diff
@@ -255,21 +255,18 @@ var _ = Describe("update_status", func() {
 					}
 				})
 
-				It("should keep NodeTaintReplacing condition", func() {
+				It("shouldn't keep NodeTaintReplacing condition", func() {
 					Expect(len(processGroupStatus.ProcessGroupConditions)).To(Equal(2))
 					Expect(processGroupStatus.GetCondition(fdbv1beta2.NodeTaintDetected)).NotTo(BeNil())
 					Expect(processGroupStatus.GetCondition(fdbv1beta2.NodeTaintReplacing)).NotTo(BeNil())
 
 					// Remove the node taint
-					node.Spec.Taints = []corev1.Taint{}
-					err = k8sClient.Update(context.TODO(), node)
-					Expect(err).NotTo(HaveOccurred())
+					node.Spec.Taints = nil
+					Expect(k8sClient.Update(context.TODO(), node)).NotTo(HaveOccurred())
 					globalControllerLogger.Info("Remove node taint", "Node name", pod.Name, "Node taints", node.Spec.Taints, "Now", time.Now())
 
-					err = validateProcessGroup(context.TODO(), clusterReconciler, cluster, pod, pvc, pod.ObjectMeta.Annotations[fdbv1beta2.LastConfigMapKey], processGroupStatus, cluster.IsTaintFeatureDisabled(), logger)
-					Expect(err).NotTo(HaveOccurred())
-					Expect(len(processGroupStatus.ProcessGroupConditions)).To(Equal(1))
-					Expect(processGroupStatus.GetCondition(fdbv1beta2.NodeTaintReplacing)).NotTo(Equal(nil))
+					Expect(validateProcessGroup(context.TODO(), clusterReconciler, cluster, pod, pvc, pod.ObjectMeta.Annotations[fdbv1beta2.LastConfigMapKey], processGroupStatus, cluster.IsTaintFeatureDisabled(), logger)).NotTo(HaveOccurred())
+					Expect(processGroupStatus.ProcessGroupConditions).To(BeEmpty())
```

Comment:
change is because we are hitting that limit now? I am not entirely clear what would cause the change other than if it is hitting the fault domain cap due to the test setup?